src/Core/Tool/Signer.php

```diff
@@ -27,42 +27,19 @@ public function sign($fullUrl, $json)
     public function computeSignature($url, $json)
     {
         $url = $url . $json;
+
         // This function calculates the HMAC hash of the data with the key
         // passed in
         // Note: hash_hmac requires PHP 5 >= 5.1.2 or PECL hash:1.1-1.5
         // Or http://pear.php.net/package/Crypt_HMAC/
         return base64_encode(hash_hmac("sha256", $url, $this->authToken, true));
     }
 
-    public function validate($expectedSignature, $url, $data = array())
+    public function validate($expectedSignature, $url, $data = "")
     {
-        return self::compare(
-            $this->computeSignature($url, json_encode($data)),
```

Comment:
@kinstelli note I dropped the `json_encode` here... so we expect request body as a string